WebRoot/js/zimbraMail/share/model/ZmSettings.js

```diff
@@ -1010,7 +1011,7 @@ function() {
 	this.registerSetting("SEARCH_ENABLED",					{type:ZmSetting.T_COS, dataType:ZmSetting.D_BOOLEAN, defaultValue:true});
 	this.registerSetting("SHORTCUT_LIST_ENABLED",			{type:ZmSetting.T_COS, dataType:ZmSetting.D_BOOLEAN, defaultValue:true});
 	this.registerSetting("OFFLINE_ENABLED",					{type:ZmSetting.T_COS, dataType:ZmSetting.D_BOOLEAN, defaultValue:appCtxt.isOffline});
-	this.registerSetting("SPELL_CHECK_ENABLED",				{type:ZmSetting.T_COS, dataType:ZmSetting.D_BOOLEAN, defaultValue:!appCtxt.isOffline && (!AjxEnv.isSafari || AjxEnv.isSafari3up || AjxEnv.isChrome)});
+	this.registerSetting("SPELL_CHECK_ENABLED",				{type:ZmSetting.T_COS, dataType:ZmSetting.D_BOOLEAN});
```

Comment:
Can you let me know the reason why the default value was removed here and added to ZmAppCtxt.js?

Reply:
The default will be overridden by value coming from backend. So moved this value to appCtxt to take the browser and offline values into consideration while enabling/disabling the button.

Reply:
I understood.
`appCtxt.get(ZmSetting.SPELL_CHECK_ENABLED)` is used in ZmComposeView as well. I think `appCtxt.isSpellCheckerAvailable()` can be used.
```
$ grep -r "appCtxt.get(ZmSetting.SPELL_CHECK_ENABLED)" WebRoot/js/
WebRoot/js/zimbraMail/core/ZmAppCtxt.js:        if (!appCtxt.get(ZmSetting.SPELL_CHECK_ENABLED)) {
WebRoot/js/zimbraMail/mail/view/ZmComposeView.js:       if ((!isDraft || forceBail) && appCtxt.get(ZmSetting.SPELL_CHECK_ENABLED) &&
```

Reply:
updated